test/modules/project.service.js

```diff
@@ -16,178 +16,105 @@ const git = require('simple-git/promise');
-) {
-    const completeOptions = completeCreationOptions(options);
-    await createProject(completeOptions, expectedResStatus, awaitSocketConfirmation);
-    await awaitProject(completeOptions.name);
-    const projectID = await getProjectIdFromName(completeOptions.name);
-    return projectID;
-}
-
 /**
- * DEPRECATED - remove from here when removed from all test files
  * @param {JSON} [options] e.g. { name: 'example' }
- * @param {number} [expectedResStatus] default 202
- * @param {boolean} [awaitSocketConfirmation] false by default, so won't wait for projectStart. Set to true to make it wait until the project is starting
+ * @param {number} [expectedResStatus] default 200
  */
-async function createProject(
-    options,
-    expectedResStatus = 202,
-    awaitSocketConfirmation = false,
-) {
-    const completeOptions = completeCreationOptions(options);
-    const req = () => reqService.chai
-        .post('/api/v1/projects')
-        .set('Cookie', ADMIN_COOKIE)
-        .send(completeOptions);
-    awaitSocketConfirmation
-        ? await reqService.makeReqAndAwaitSocketMsg(req, expectedResStatus, { msgType: 'projectStarting' })
-        : await reqService.makeReq(req, expectedResStatus);
-}
-
-/**
- * DEPRECATED - remove from here when removed from all test files
- * @param {JSON} [options] e.g. { name: 'example' }
- * @param {number} [expectedResStatus] default 202
- */
-async function createNonBuiltProject(
-    options,
-    expectedResStatus = 202,
-) {
-    const name = options.name;
-    const completeOptions = completeCreationOptions(options);
-    completeOptions.autoBuild = false;
-    const req = () => reqService.chai
-        .post('/api/v1/projects')
-        .set('Cookie', ADMIN_COOKIE)
-        .send(completeOptions);
-    await reqService.makeReqAndAwaitSocketMsg(req, expectedResStatus, { name, msgType: 'projectCreation' });
-    const projectID = await getProjectIdFromName(completeOptions.name);
-    return projectID;
-}
-
-/**
- * DEPRECATED - remove from here when removed from all test files
- * @param {JSON} [options]
- * @returns {JSON} JSON with 'name', 'language', and (if 'language' is Java) 'framework' fields set to default values if they were empty
- */
-function completeCreationOptions(options = {}) {
-    if (!isObject(options)) throw new Error(`'${options}' should be an object`);
-    if (options.extension) return options;
-
-    const completeOptions = { ...options };
-
-    completeOptions.name = options.name || generateUniqueName();
-    completeOptions.language = templateOptions[options.type].language;
-    completeOptions.url = templateOptions[options.type].url;
-    
-    return completeOptions;
-}
+async function bindProject(options) {
+    const resFromBindStart = await bindStart(options);
+    const { projectID } = resFromBindStart.body;
+    await uploadFiles(projectID, options.path);
+    const resFromBindEnd = await bindEnd(projectID);
+    awaitProjectBuilding(projectID);
```

Comment:
For just binding a project do we need to wait for the project to start building? Even if something in the project build goes wrong I'd say we have still successfully bound a project?